index.d.ts

```diff
@@ -643,7 +643,8 @@ declare namespace Eris {
       token?: string,
       reason?: string,
     ): Promise<Webhook>;
-    public executeWebhook(webhookID: string, token: string, options: WebhookPayload): Promise<void>;
+    public executeWebhook(webhookID: string, token: string, options: WebhookPayload & { wait: true; }): Promise<Message>;
+    public executeWebhook(webhookID: string, token: string, options: WebhookPayload & { wait?: false; ): Promise<void>;
```

Comment:
Missing `}`?

Also, does the `wait?` mean that it'll also match an undefined `wait?

Reply:
`wait?` means the key can be missing, it can't be undefined if it's present.

Reply:
```suggestion
    public executeWebhook(webhookID: string, token: string, options: WebhookPayload): Promise<void>;
```

The `wait: true` is useful as an overload but the base should just default back to the WebhookPayload. This will also allow users to input `undefined` and wait will be set as the `wait?: boolean` allowing it to accept true and false as well.

Reply:
You have to include the false, else it will complain that false is not assignable to true.

Reply:
You can leave it as is, as discussed in the discord channel keeping it strictly expressed in this way is preferred.

The `false`, `true`, and `undefined` will all be inherited from WebhookPayload. The ?: false is redundant code imo.

Reply:
@Skillz4Killz You have to leave the false, else it will complain that [false is not assignable to true](https://butts-are.cool/Code_-_Insiders_KrU6Zyung1.png), as well as [the property "wait" being missing if not included](https://butts-are.cool/Code_-_Insiders_a6jmNLPR2j.png)